src/transformers/trainer.py

```diff
@@ -468,7 +473,11 @@ def setup_wandb(self):
             logger.info(
                 'Automatic Weights & Biases logging enabled, to disable set os.environ["WANDB_DISABLED"] = "true"'
             )
-            combined_dict = {**self.model.config.to_dict(), **self.args.to_sanitized_dict()}
+            try:
+                combined_dict = {**self.model.config.to_dict(), **self.args.to_sanitized_dict()}
+            except AttributeError:
+                # in case the model has no config
+                combined_dict = {**self.args.to_sanitized_dict()}
```

Comment:
Is there an example of a model without a configuration?

Reply:
Ah yes, it's something @sgugger mentioned as well - when writing for `Trainer` we assume that the model is a `PretrainedModel`, but the one in the test doesn't inherit from `PretrainedModel` which is why I put this in. @julien-c also liked the idea of `Trainer` being domain-agnostic (eg not only NLP for example) so I figured might as well put this line in since it isn't expensive.